recipes/cigi-cl/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import is_msvc, is_msvc_static_runtime
+from conan.tools.apple import is_apple_os
+import os
+
+required_conan_version = ">=1.53.0"
+
+class CigiClConan(ConanFile):
+    name = "cigi-cl"
+    description = "Industry standard communication with compliant image generators"
+    license = "LGPL-2.1-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/cigi/"
+    topics = ("Simulation", "Interface Engines", "Data Visualization")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 98
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if is_apple_os(self):
```

Comment:
This is the error while trying to build for *MacOS*:
```
CigiOutgoingMsg.cpp:196:38: error: use of undeclared identifier 'PTHREAD_MUTEX_RECURSIVE_NP'
pthread_mutexattr_settype( &attr, PTHREAD_MUTEX_RECURSIVE_NP );
```

As I mentioned [here](https://github.com/conan-io/conan-center-index/pull/24810#issuecomment-2267611084):
> Apparently the `_NP` in `PTHREAD_RECURSIVE_MUTEX_NP` means a _non-portable_ extension, which is not supported by Apple as stated [here](https://stackoverflow.com/questions/53075327/use-of-undeclared-identifier-pthread-recursive-mutex-initializer-np/53075997#53075997). An inspection of the [code](https://opensource.apple.com/source/libpthread/libpthread-454.80.2/include/pthread/pthread.h.auto.html) shows it's indeed not provided. Therefore, updated recipe so that `validate()` will report Apple is not supported.

I don't think it's an official limitation. Probably just wasn't properly tested.

Reply:
That's totally okay. I updated the recipe to clarify it better, so anyone can understand why that invalid configuration is there.